php-checks/src/main/java/org/sonar/php/checks/regex/AbstractRegexCheck.java

```diff
@@ -83,11 +83,13 @@
   protected void checkFunctionCall(FunctionCallTree tree) {
     CheckUtils.argumentValue(tree, "pattern", 0)
       .flatMap(AbstractRegexCheck::getLiteral)
+      .filter(this::hasValidDelimiters)
```

Comment:
Yeah. Before the check for valid delimiters was hidden in the getLiteralMethod. Do extend the AbstractRegexCheck logic for S6393. I wanted to prevent duplicating too much code by separating this logic.